src/newrelic.coffee

```diff
@@ -62,20 +71,63 @@ plugin = (robot) ->
       .header('Content-Type','application/x-www-form-urlencoded')
 
   get = (path, cb) ->
-    return cb({message:'HUBOT_NEWRELIC_API_KEY is not set'}) if !apiKey
     _request(path).get() _parse_response(cb)
+
   post = (path, data, cb) ->
-    return cb({message:'HUBOT_NEWRELIC_API_KEY is not set'}) if !apiKey
     _request(path).post(data) _parse_response(cb)
 
-  send_message = (msg, messages) ->
-    if messages.length < 1000
+
+  send_message = (msg, messages, longMessageIntro="") ->
+    if messages.length < maxMessageLength
       msg.send messages
     else
       gist {content: messages, enterpriseOnly: true, fileExtension: 'md'}, (err, resp, data) ->
-        url = data.html_url
-        msg.send "View output at: " + url
+        msg.send "#{longMessageIntro} View output at: #{data.html_url}"
+
+  message_room = (robot, room, messages, longMessageIntro="") ->
+    if messages.length < maxMessageLength
+      robot.messageRoom room, messages
+    else
+      gist {content: messages, enterpriseOnly: true, fileExtension: 'md'}, (err, resp, data) ->
+        robot.messageRoom room, "#{longMessageIntro} View output at: #{data.html_url}"
 
+  poll_violations = (robot) ->
+    get "alerts_violations.json?only_open=true", (err, json) ->
+      if err
+        robot.messageRoom room, "New Relic Violations Polling Failed: #{err.message}"
+      else
+        #console.log json.violations
+        console.log "New Relic alerts poll. #{json.violations.length} alert(s) found"
+
+        previous = robot.brain.data.newrelicdata.violations
```

Comment:
Is it necessary to nest the data under the `violations` property? Can you just store it under a key like `newrelicviolations`?

i.e. `previous = robot.brain.get 'newrelicviolations'`